packages/expo-updates/ios/EXUpdates/AppController.swift

```diff
@@ -3,19 +3,92 @@
+  let launchedUpdate: Update?
+  let embeddedUpdate: Update?
+  let isEmergencyLaunch: Bool
+  let isEnabled: Bool
+  let releaseChannel: String
+  let isUsingEmbeddedAssets: Bool
+  let runtimeVersion: String?
+  let checkOnLaunch: CheckAutomaticallyConfig
+  let requestHeaders: [String: String]
+  let assetFilesMap: [String: Any]?
+  let isMissingRuntimeVersion: Bool
+}
+
+public typealias AppRelaunchCompletionBlock = (_ error: Error?) -> Void
+
+public enum FetchUpdateResult {
+  case success(manifest: [String: Any])
+  case failure
+  case rollBackToEmbedded
+  case error(error: Error)
+}
+
+@objc(EXUpdatesAppController)
+public protocol IPublicAppController {
+  /**
+   The RCTBridge for which EXUpdates is providing the JS bundle and assets.
+   This is optional, but required in order for `Updates.reload()` and Updates module events to work.
+   */
+  @objc weak var bridge: AnyObject? { get set }
+
+  /**
+   Delegate which will be notified when EXUpdates has an update ready to launch and
+   `launchAssetUrl` is nonnull.
+   */
+  @objc weak var delegate: AppControllerDelegate? { get set }
+
+  /**
+   The URL on disk to source asset for the RCTBridge.
+   Will be null until the AppController delegate method is called.
+   This should be provided in the `sourceURLForBridge:` method of RCTBridgeDelegate.
+   */
+  @objc func launchAssetUrl() -> URL?
 
-public typealias AppRelaunchCompletionBlock = (Bool) -> Void
+  @objc var isStarted: Bool { get }
+  @objc func start()
+}
+
+public protocol IAppController: IPublicAppController {
+  func getConstantsForModule() -> UpdatesModuleConstants
+  func requestRelaunch(
+    success successBlockArg: @escaping () -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+  func checkForUpdate(
+    success successBlockArg: @escaping (_ remoteCheckResult: RemoteCheckResult) -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+  func fetchUpdate(
+    success successBlockArg: @escaping (_ fetchUpdateResult: FetchUpdateResult) -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+  func getExtraParams(
+    success successBlockArg: @escaping (_ extraParams: [String: String]?) -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+  func setExtraParam(
+    key: String,
+    value: String?,
+    success successBlockArg: @escaping () -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+  func getNativeStateMachineContext(
+    success successBlockArg: @escaping (_ stateMachineContext: UpdatesStateContext) -> Void,
+    error errorBlockArg: @escaping (_ error: Exception) -> Void
+  )
+}
 
 @objc(EXUpdatesAppControllerDelegate)
 public protocol AppControllerDelegate: AnyObject {
-  func appController(_ appController: AppController, didStartWithSuccess success: Bool)
+  func appController(_ appController: IPublicAppController, didStartWithSuccess success: Bool)
```

Comment:
Not sure about the Android style name. In looking around our codebase, maybe `AppControllerInterface` would be better.

Reply:
Or `UpdatesControllerInterface` if it will be accessed outside `expo-updates` module.